src/concurrency/timestamp_ordering_transaction_manager.cpp

```diff
@@ -778,130 +778,136 @@ ResultType TimestampOrderingTransactionManager::CommitTransaction(
   oid_t database_id = 0;
   if (static_cast<StatsType>(settings::SettingsManager::GetInt(settings::SettingId::stats_mode)) !=
       StatsType::INVALID) {
-    if (!rw_set.empty()) {
-      database_id =
-          manager.GetTileGroup(rw_set.begin()->first)->GetDatabaseId();
+    if (!rw_set.IsEmpty()) {
+      // Call the GetConstIterator() function to explicitly lock the cuckoohash
+      // and initilaize the iterator
```

Comment:
If `CommitTransaction` is single-threaded and there isn't any concurrent access to the read-write set, is locking the table necessary?

Reply:
Cukoohash only provides an iterator via the lock_table interface. This is because there's no way to tell cuckoohash that nobody else would be accessing the table while the iterator is in context. I agree that it seems kind of inefficient but in essence it just does number of bucket atomic adds. (In the default case this would be 64/4 = 16)